rfc/parameterizedMetrics.md

```diff
@@ -0,0 +1,236 @@
+
+So
+```
+"metricName(parameterName=parameterValue)"
+```
+
+will result in a dynamic metric based on `metricName`, but substituting the parameter value when calculating it.  Also, the above string will be used as the column name in the resulting report response schema. 
+
+Motivation
+----------
+
+Standard metrics in Fili are configured at load time.  However if reporting requires many different similar caluclations, such as adjusting precision or converting values of different units into a shared aggregate, or applying custom metric filters, it may be impractical or even impossible to configure all the interesting report metrics.
+
+With parameterized metrics, open-ended variations may be supported for a given conceptual metric calculation, simplifying configuration and possibly unlocking otherwise impractical freedom in metric calculation.
+
+
+The structure of a parameterized metric:
+----------------------------------------
+
+```
+metricName(parameterName=value)
+
+metricName [ '(' parameterName=value [ ',' parameterName=value ]*  ')' ]?
+```
+
+**metricName** -> an api metric name which should appear as a resource at the `/v1/metrics/metricName` endpoint.
+
+**parameterName** -> an identifier for a parameter supported by this metric.  Identifier names should follow the same format rules as api resource identifiers.
+
+**value** -> a value to be used for creating a modified version of baseMetric.  URL encoding may be required if some of the characters in the value contain special characters.  Escaping values containing ampersand is mandatory and it may also be necessary to encode values containing comma or parentheses depending on implementation.
+
+Values containing characters delimiters such as comma or parantheses must have them nested in other scoping characters such as quotes, square brackets or parantheses.
+
+If multiple parameter names are used for this metric they should be joined by commas with no intermediate spaces (similar to the filters value syntax)
+
+i.e.
+
+```
+metricName(parameterName=value,otherParameter=otherValue)
+```
+
+
+Response schema
+---------------
+
+Unless aliasing is supported and used (see below), responses will use the full string of the metric expression as the column name for values resulting from parametrized metric evaluation.
+
+Example query:
+```
+...&metrics=revenue,netRevenue,revenue(currency=USD)
+```
+
+Example response:
+```json
+{
+  "rows": [
+    {
+      "dateTime": "2018-02-01 00:00:00.000",
+      "mediaType|id": "-2",
+      "currency|id": "AUD",
+      "revenue": "100",
+      "revenue|currency": "AUD",
+      "netRevenue": "50",
+      "netRevenue|currency": "CAD",
+      "revenue(currency=USD)": "120",
+      "revenue(currency=USD)|currency": "USD"
+      
+    }
+    ]
+}
+```
+
+Reserved parameter names
```

Comment:
Are there any other parameter names we want to cordon off as "reserved words" for future use? It's much harder to add new parameters defined into the system later. To put it a different way: Are there any other parameters that we can foresee applying to _all_ metrics?

Or perhaps we want to reserve a "prefix" as a namespace for future core additions?

Reply:
I like this.  So 'core:X' would be reserved and 'X' would be the default alias for that namespaced term?  I like that. It makes things flexible and could let us strip the filter and as cases out of this spec.